feature-libs/order/occ/order-occ.module.ts

```diff
@@ -79,6 +85,30 @@ import { defaultOccOrderConfig } from './config/default-occ-order-config';
       useExisting: OccReorderOrderNormalizer,
       multi: true,
     },
+    {
+      provide: OrderHistoryAdapter,
+      useFactory: (
+        httpClient: HttpClient,
+        occEndpointsService: OccEndpointsService,
+        converterService: ConverterService
+      ) => {
+        const enhancedUI = inject(MYACCOUNT_ORDER_ENHANCED_UI);
+        if (enhancedUI) {
+          return new OccOrderHistoryExtendedAdapter(
+            httpClient,
+            occEndpointsService,
+            converterService
+          );
+        } else {
+          return new OccOrderHistoryAdapter(
+            httpClient,
+            occEndpointsService,
+            converterService
+          );
+        }
+      },
+      deps: [HttpClient, OccEndpointsService, ConverterService],
+    },
```

Comment:
[MAJOR]
It's not recommended to instantiate classes manually with `new`, but instead rely on the Angular Dependency injection.

Did you consider simply injecting the original instances conditionally?
```suggestion
    OccOrderHistoryAdapter,
    OccOrderHistoryExtendedAdapter,
    {
      provide: OrderHistoryAdapter,
      useFactory: () => {
        const enhancedUI = inject(MYACCOUNT_ORDER_ENHANCED_UI);
        return enhancedUI
          ? inject(OccOrderHistoryExtendedAdapter)
          : inject(OccOrderHistoryAdapter);
      },
    },
```

Reply:
done